File: api/src/main/java/com/ruby/rubia_server/core/dto/UserAIAgentDTO.java
package com.ruby.rubia_server.core.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserAIAgentDTO {

    private UUID id;
    private UUID companyId;
    private String companyName;
    private UUID userId;
    private String userName;
    private UUID aiAgentId;
    private String aiAgentName;
    private Boolean isDefault;
    private LocalDateTime assignedAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/DTOs/UserAIAgentDto.cs
using System.ComponentModel.DataAnnotations;

namespace Rubia.Server.DTOs;

public class UserAIAgentDto
{
    public Guid Id { get; set; }
    public Guid CompanyId { get; set; }
    public string? CompanyName { get; set; }
    public Guid UserId { get; set; }
    public string? UserName { get; set; }
    public Guid AiAgentId { get; set; }
    public string? AiAgentName { get; set; }
    public bool IsDefault { get; set; }
    public DateTime AssignedAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class CreateUserAIAgentDto
{
    [Required(ErrorMessage = "Company ID is required")]
    public Guid CompanyId { get; set; }

    [Required(ErrorMessage = "User ID is required")]
    public Guid UserId { get; set; }

    [Required(ErrorMessage = "AI Agent ID is required")]
    public Guid AiAgentId { get; set; }

    public bool IsDefault { get; set; } = false;
}

public class UpdateUserAIAgentDto
{
    public Guid? AiAgentId { get; set; }
    public bool? IsDefault { get; set; }
}